File: app/src/main/java/ru/nikshlykov/englishwordsapp/Link.java
package ru.nikshlykov.englishwordsapp;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.Index;

import static androidx.room.ForeignKey.CASCADE;

@Entity(tableName = "Links", foreignKeys = {
        @ForeignKey(
                entity = Subgroup.class,
                parentColumns = "_id",
                childColumns = "SubgroupId",
                onDelete = CASCADE,
                onUpdate = CASCADE),
        @ForeignKey(
                entity = Word.class,
                parentColumns = "_id",
                childColumns = "WordId",
                onDelete = CASCADE,
                onUpdate = CASCADE)},
        primaryKeys = {"SubgroupId", "WordId"},
        indices = {
                @Index("WordId"),
                @Index("SubgroupId")})
public class Link {

    @NonNull
    @ColumnInfo(name = "SubgroupId")
    public long subgroupId;

    @NonNull
    @ColumnInfo(name = "WordId")
    public long wordId;

    @NonNull
    @ColumnInfo(name = "LevelInParentGroup", defaultValue = "10")
    public int levelInParentGroup;
}

File: data/src/main/java/ru/nikshlykov/data/database/models/Link.kt
package ru.nikshlykov.data.database.models

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index

@Entity(
  tableName = "Links",
  foreignKeys = [ForeignKey(
    entity = Subgroup::class,
    parentColumns = ["_id"],
    childColumns = ["SubgroupId"],
    onDelete = ForeignKey.CASCADE,
    onUpdate = ForeignKey.CASCADE
  ), ForeignKey(
    entity = Word::class,
    parentColumns = ["_id"],
    childColumns = ["WordId"],
    onDelete = ForeignKey.CASCADE,
    onUpdate = ForeignKey.CASCADE
  )],
  primaryKeys = ["SubgroupId", "WordId"],
  indices = [Index("WordId"), Index("SubgroupId")]
)
class Link(
  @ColumnInfo(name = "SubgroupId") val subgroupId: Long,
  @ColumnInfo(name = "WordId") val wordId: Long
) {

  @ColumnInfo(name = "LevelInParentGroup", defaultValue = "10")
  var levelInParentGroup = 0
}